server/src/main.rs

```diff
@@ -1,65 +1,103 @@
-use anyhow::Context;
-use axum::{http::{Method, HeaderValue}, Extension};
+use axum::{http::Method, Extension};
 use clap::Parser;
-use osc_api::{router::Build, Config};
-use sqlx::postgres::PgPoolOptions;
+use std::net::SocketAddr;
 use std::sync::Arc;
+use tokio::signal;
 use tower::ServiceBuilder;
-use tower_http::cors::{CorsLayer, Any};
+use tower_http::{
+    cors::{Any, CorsLayer},
+    trace::TraceLayer,
+};
 
-extern crate dotenv;
+use crate::config::{Config, Context};
+use crate::db::cache::CacheManager;
+use crate::db::DBConnectionManager;
+use crate::prelude::*;
+use crate::router::Build;
+
+mod config;
+mod db;
+mod error;
+mod mina;
+mod prelude;
+mod router;
+mod schema;
+
+extern crate tracing;
+
+pub(crate) const MINA_GOVERNANCE_SERVER: &str = "mina_governance_server";
 
 #[tokio::main]
-async fn main() -> anyhow::Result<()> {
-    dotenv::dotenv().ok();
-    env_logger::init_from_env(env_logger::Env::new().default_filter_or("info"));
+async fn main() -> Result<()> {
+    dotenvy::dotenv().ok();
+    config::init_tracing();
 
     let config = Config::parse();
-    let ledger_cache = osc_api::ledger::LedgerCache::builder()
-        .time_to_live(std::time::Duration::from_secs(60 * 60 * 12))
-        .build();
+    let cache = CacheManager::build();
 
-    let signal_cache = osc_api::signal::SignalCache::builder()
-        .time_to_live(std::time::Duration::from_secs(60 * 3))
-        .build();
+    tracing::info!(
+        target: MINA_GOVERNANCE_SERVER,
+        "Initializing database connection pools..."
+    );
 
-    let mainnet_db = PgPoolOptions::new()
-        .max_connections(25)
-        .connect(&config.database_url)
-        .await
-        .context("Error: Could not connect to mainnet database.")?;
-
-    let allowed_origins = match std::env::var("ALLOWED_ORIGINS") {
-        Ok(val) => {
-            let origin = val.split(",").map(|s| s.trim()).collect::<Vec<&str>>().join(",");
-            HeaderValue::from_str(&origin).unwrap()
-        },
-        Err(_) => return Err("Origin not allowed"),
-    };
+    let conn_manager = DBConnectionManager::get_connections(&config);
+
+    let router = axum::Router::build().layer(
+        ServiceBuilder::new()
+            .layer(TraceLayer::new_for_http())
+            .layer(
+                CorsLayer::new()
+                    .allow_methods([Method::GET, Method::POST, Method::OPTIONS])
+                    .allow_origin(Any)
```

Comment:
This overwrote Jason's changes. This still should be configurable via environment variable but fall back ANY if not is provided

Reply:
Correction. We should throw an error if no allowed_origins value is provided